modules/task/backend.go

```diff
@@ -78,31 +74,14 @@ func (b NopBackend) IsRunning() bool {
 type inMemBackend struct {
 	bufQueue  chan []byte
 	isRunning bool
+	isStopped bool
```

Comment:
There are 3 states,
Not started -> isRunning = false, isStopped = false
Started/Running -> isRunning = true, isStopped = false
Stopped -> isRunning = false, isStopped = true

I added this extra bool after @alsamylkin 's  comment to add validation for cases when someone calls Start after Stop. In previous code, it would panic/timeout on a channel. He was recommending a clearer error message. With isRunning, this is hard to distinguish because it's false both before the service started and after the service has been stopped.